lms/services/organization.py

```diff
@@ -65,10 +65,14 @@ def auto_assign_organization(
                     "Multiple organization matches found for application instance %s",
                     application_instance.id,
                 )
+
             org = orgs[0]
 
         else:
             org = Organization()
+            self._db_session.add(org)
+            # Get org.id and org._public_id from the DB
+            self._db_session.flush()
```

Comment:
This doesn't seem to be a problem on other code path and only affecting the non-canvas configuration. 

Some other implicit flush might be hiding the problem in other cases.

Reply:
The comment here is a little misleading, as we get the id from the DB, but Python is writing the `_public_id` to the DB.

Maybe `Ensure we have ids`?